docs/source/advanced/language-models.rst

```diff
@@ -150,20 +249,206 @@ To co-train BERT and GNN models, we need to add one more argument, ``--lm-train-
 
-.. note:: It will take longer time to co-train BERT and GNN models compared to no co-train.
+Model performance
+#####################
 
-Only Use BERT Models
-------------------------
-GraphStorm also allows users to only use BERT models to perform graph tasks. We can add another argument, ``--lm-encoder-only``, to control whether only use BERT models or not.
+**Run time:**
+
+* Link prediction fine-tuning training: 51s per epoch
+* Link prediction fine-tuned model saving: 1s each time
+
+* GNN training: 2s per epoch
+* GNN validation: 0.5s per epoch
+* GNN model saving: 2~3s each time
+
+**Best accuracy on validation set:**
+
+* RGCN: ~0.64
+* RGAT: ~0.65
+* HGT: ~0.63
+
+All above best accuracy values were achieved using LP fine-tuning.
+
+2.3 Auto two-step co-training (GLEM)
+`````````````````````````````````````
+
+There are two important pre-requisites for achieving good performance when using GLEM strategy.
+
+1. Well pre-trained LM and GNN before the GLEM co-training: empirically, LM or GNN models that are not well-trained lead to degraded performance when co-training with GLEM directly. Therefore, users need to pre-train the LMs and GNN Models first.
+
+To pre-train LMs and GNN models, users can follow the :ref:`22two_step_mannually` instruction and save the best LMs and GNN models.
+
+2. The pseudolabeling technique: this technique predicts pseudolabels on the unlabeled nodes and uses them as additional supervision signals for mutual distillation between LM and GNN. This can be enabled by the setting the ``use_pseudolabel`` argument.
+
+GraphStorm configurations
+##########################
+
+To use GLEM, users need to set a new configuration se, called ``training_method``, which specifies how to utilize specific model training method. Users can refer to the ``acm_glem_nc_pretrain.yaml`` that includes the following ``training_method`` related configurations.
+
+.. code-block:: yaml
+
+    training_method:
+      name: glem
+      kwargs:
+        em_order_gnn_first: false
+        inference_using_gnn: true
+        pl_weight: 0.5
+        num_pretrain_epochs: 100
+    use_pseudolabel: true
+
+Within the ``traing_method`` section, there are two important configurations. First, the ``pl_weight`` defines the weights of pseudolabel, which determines the importance of pseudolabel. Users can lower the value to reduce the influence of using pseudolabel. The second important configuration is the ``num_pretrain_epochs``. The GLEM method provides its own pre-training implementation, which train LMs and GNN models iteratively in one epoch, i.e., first fix GNN model and train LMs in one forward and backward loop, and then fix LM but use it to embed text as input for GNN models to be trained in one loop. In the pre-training epochs, GLEM will not use the pseudolabel, but the true labels only, even if users set the ``use_pseudolabel`` configuration to be true.
```

Comment:
"importance of pseudolabel" -> importance of pseudolabel in mutual distillation

Reply:
changed.